### PR TITLE
MINOR: stateful docs for aggregates

### PR DESCRIPTION
**Pull request by enothereska**

### CODE REVIEW

**Comment by enothereska** [comment]
Not done yet, WiP

**Comment by enothereska** [comment]
@dguy this next section is ready. Thanks.

File: docs/streams/developer-guide.html [hunk 1]
```diff
@@ -1014,13 +1014,819 @@
 
 
     <h5><a id="streams_dsl_transformations_stateful" href="#streams_dsl_transformations_stateful">Stateful transformations</a></h5>
-    Stateless transformations, by definition, do not depend on any state for processing, and hence implementation-wise they do not
-    require a state store associated with the stream processor.Stateful transformations, by definition, depend on state for processing
-    inputs and producing outputs, and hence implementation-wise they require a state store associated with the stream processor. For
-    example, in aggregating operations, a windowing state store is used to store the latest aggregation results per window; in join
-    operations, a windowing state store is used to store all the records received so far within the defined window boundary.
+    <h6><a id="treams_dsl_transformations_stateful_overview" href="#streams_dsl_transformations_stateful_overview">Overview</a></h6>
```
Comment (dguy) [inline docs/streams/developer-guide.html]: treams -> streams

File: docs/streams/developer-guide.html [hunk 2]
```diff
@@ -1014,13 +1014,819 @@
...
+        <li><a href="#streams_dsl_joins">Joining</a>
+        <li><a href="#streams_dsl_windowing">Windowing (as part of aggregations and joins)</a>
+        <li>Applying custom processors and transformers, which may be stateful, for Processor API integration
+    </ul>
+    </p>
+    <p>
+        The following diagram shows their relationships:
+    </p>
+    <figure>
+        <img class="centerd" src="/{{version}}/images/streams-stateful_operations.png" style="width:500pt;">
```
Comment (dguy) [inline docs/streams/developer-guide.html]: centered

File: docs/streams/developer-guide.html [hunk 3]
```diff
@@ -1014,13 +1014,819 @@
...
+        defined window boundary.
+    </p>
+    <p>
+        Note, that state stores are fault-tolerant. In case of failure, Kafka Streams guarantees to fully restore
+        all state stores prior to resuming the processing.
+    </p>
+    <p>
+        Available stateful transformations in the DSL include:
+    <ul>
+        <li><a href=#streams_dsl_aggregations>Aggregating</a>
```
Comment (dguy) [inline docs/streams/developer-guide.html]: closing `</li>` tags missing here and elsewhere

Comment (enothereska) [reply docs/streams/developer-guide.html]: > An li element's end tag may be omitted if the li element is immediately followed by another li element or if there is no more content in the parent element. https://www.w3.org/TR/2014/REC-html5-20141028/syntax.html#optional-tags

Comment (enothereska) [reply docs/streams/developer-guide.html]: I'll add them anyway.

**Comment by enothereska** [comment]
Thanks @dguy

**Review by dguy — APPROVED** [review]
Thanks @enothereska, LGTM

**Pull request closed without merging**